lib/game.rb

```diff
@@ -8,44 +8,70 @@
 
 
 class Game
+
+  attr_reader :row, :column
+
   HARDCODED_MODE = 1
-  COMPUTER_MODE = 2
-  HUMAN_MODE = 3
+  COMPUTER_PLACES_SHIPS = 2
+  HUMAN_PLACES_SHIPS = 3
+  HUMAN_PLAYER = 4
+  COMPUTER_PLAYER = 5
+
   def initialize(display, board)
     @display = display
     @board = board
   end
 
-  def play(ship_mode)
+  def play(ship_mode, player_mode)
     welcome
     place_ships_on_grid(ship_mode)
     present_board(@board.grid)
     until @board.hit_all?(@board.grid)
-      turn(@board.grid)
+      turn(player_mode, @board.grid)
     end
     goodbye
   end
 
   def place_ships_on_grid(ship_mode)
     if ship_mode == HARDCODED_MODE
       @board.grid_with_one_hardcoded_ship
-    elsif ship_mode == COMPUTER_MODE
+    elsif ship_mode == COMPUTER_PLACES_SHIPS
       validator = CoordinateValidator.new(@board)
       computer_player = ComputerPlayer.new(@board, validator)
       computer_player.generate_ships
-    elsif ship_mode == HUMAN_MODE
+    elsif ship_mode == HUMAN_PLACES_SHIPS
       human_player = HumanPlayer.new(@board, @display)
       human_player.generate_ships
     end
   end
 
-  def turn(grid)
-    get_coordinates
+  def turn(player_mode, grid)
+    if player_mode == HUMAN_PLAYER
```

Comment:
how can this be refactored polymorphically?  `get_coordinates`/`get_computer_coordinates` should be the player's concerns.
THe `present_marked_board` call is repeated too so can just be called once if the sleep is moved into ComputerPlayer